pkg/pod/pods.go

```diff
@@ -971,10 +971,18 @@ func (p *Pod) AppImageManifest(appName string) (*schema.ImageManifest, error) {
 // CreationTime returns the time when the pod was created.
 // This happens at prepare time.
 func (p *Pod) CreationTime() (time.Time, error) {
-	if p.isPrepared || p.isRunning() || p.AfterRun() {
-		return p.getModTime("pod")
+	if !(p.isPrepared || p.isRunning() || p.AfterRun()) {
+		return time.Time{}, nil
+	}
+	t, err := p.getModTime("pod-created")
```

Comment:
Instead of introducing a new file to check and branching code for backward compatibility here, shall we perhaps just switch to checking the ctime of a file which is actually immutable for the whole pod lifetime? The `flavor` symlink looks like a possible target here.